### PR TITLE
BUG: 23 Beta issue with timezones

### PR DESCRIPTION
**Pull request by ahgnaw**
Fixed bug 23. Beta did not work when a pandas series had a timezone and np.nan.

### CODE REVIEW

File: empyrical/stats.py [hunk 1]
```diff
@@ -665,10 +665,7 @@ def beta(returns, factor_returns, risk_free=0.0):
     """
 
     # Filter out dates with np.nan as a return value
-    indices = np.intersect1d(
-        returns.dropna().index,
-        factor_returns.dropna().index
-    )
+    indices = indices = pd.concat([returns, factor_returns]).dropna().index
```
Comment (twiecki) [inline empyrical/stats.py]: indices is assigned twice.

File: empyrical/tests/test_stats.py [hunk 2]
```diff
@@ -785,7 +785,7 @@ def test_alpha_beta_with_nan_inputs(self, returns, benchmark):
     @parameterized.expand([
         (empty_returns, simple_benchmark, np.nan),
         (one_return, one_return,  np.nan),
-        (mixed_returns, simple_benchmark, np.nan),
+        (mixed_returns, simple_benchmark, 0.44444444444444442),
```
Comment (twiecki) [inline empyrical/tests/test_stats.py]: should add a unit-test that tests with tz.

File: empyrical/stats.py (hunk 1 above)
Comment (jdricklefs) [inline empyrical/stats.py]: Double-assigned indices here.

**Comment by jbredeche** [comment]
Any performance implications here?

File: empyrical/tests/test_stats.py [hunk 3]
```diff
@@ -71,24 +71,24 @@ class TestStats(TestCase):
...
     flat_line_0 = pd.Series(
         np.linspace(0, 0, num=1000),
-        index=pd.date_range('2000-1-30', periods=1000, freq='D')
+        index=pd.date_range('2000-1-30', periods=1000, freq='D', tz='UTC')
     )
     # Flat line
     flat_line_1 = pd.Series(
         np.linspace(0.01, 0.01, num=1000),
-        index=pd.date_range('2000-1-30', periods=1000, freq='D')
+        index=pd.date_range('2000-1-30', periods=1000, freq='D', tz='UTC')
```
Comment (twiecki) [inline empyrical/tests/test_stats.py]: Sorry, I meant in addition to non-UTC. Just a single one with UTC and arbitrary data.

**Pull request closed without merging**